sdk/communication/Azure.Communication.Common/tests/Pipeline/EntraTokenGuardPolicyTests.cs

```diff
@@ -81,6 +85,60 @@ public async Task Process_WithDifferentToken_CallsProcessNext(bool async)
             }
         }
 
+        [Test]
+        [TestCase(true)]
+        [TestCase(false)]
+        public async Task Process_WithInvalidAcsToken_CallsProcessNext(bool async)
+        {
+            // Arrange
+            var policy = new EntraTokenGuardPolicy();
+            var pipeline = new Mock<HttpPipelinePolicy>();
+            var pipelines = new ReadOnlyMemory<HttpPipelinePolicy>(new HttpPipelinePolicy[] { _httpPipelinePolicyMock.Object });
+            _httpPipelinePolicyMock.Setup(m => m.ProcessAsync(It.IsAny<HttpMessage>(), It.IsAny<ReadOnlyMemory<HttpPipelinePolicy>>())).Returns(new ValueTask(Task.CompletedTask));
+            _httpPipelinePolicyMock.Setup(m => m.Process(It.IsAny<HttpMessage>(), It.IsAny<ReadOnlyMemory<HttpPipelinePolicy>>())).Verifiable();
+            var expired = DateTimeOffset.UtcNow.AddMinutes(-1);
+            var expiredAcsTokenResponse = string.Format(TokenResponseTemplate, "token", expired);
+            _httpMessage.Response = new MockResponse(200).SetContent(expiredAcsTokenResponse);
+            // Act & Assert
+            if (async)
+            {
+                await policy.ProcessAsync(_httpMessage, pipelines);
+                await policy.ProcessAsync(_httpMessage, pipelines);
+                _httpPipelinePolicyMock.Verify(p => p.ProcessAsync(_httpMessage, It.IsAny<ReadOnlyMemory<HttpPipelinePolicy>>()), Times.Exactly(2));
```

Comment:
IS this really verifying that the pipeline internally calls `ProcessNext`? At first read, it seems that it just verifies that `ProcessAsync` was called on the pipeline which happens in the two lines prior.

Reply:
So here we init `pipelines` 95th line with array of pipelines, which has only `_httpPipelinePolicyMock` pipeline. The `pipelines` is provided as a param to the `EntraTokenGuardPolicy. ProcessAsync` on 105th & 106th lines
<img width="1200" alt="image" src="https://github.com/user-attachments/assets/443dd507-4f81-4631-9bdd-067700d3340d">

inside the  `EntraTokenGuardPolicy` if the acs token invalid next pipeline is called and here we are forwarding `pipelines` param, and inside the [HttpPipelinePolicy.ProcessNext](https://github.com/Azure/azure-sdk-for-net/blob/130d1c76e36b634d8087e0746a3e68c2f0287c43/sdk/core/Azure.Core/src/Pipeline/HttpPipelinePolicy.cs#L35)  it will call the _httpPipelinePolicyMock
<img width="1023" alt="Pasted Graphic 1" src="https://github.com/user-attachments/assets/048f57de-767f-48d0-9a65-011a85c34a24">

@DominikMe would any further clarification be helpful?